### PR TITLE
Grafana/ui: Update Link component to react router 6

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**

Update Link component to react router 6

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/92567

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by JoaoSilvaGrafana — APPROVED** [review]
I've tested nav links and `TextLink`s, both seem to work. Also tested with a subpath locally because we have had problems in the past and seems good! 👏

**Review by eledobleefe — APPROVED** [review]
I've played a bit with several `Link` components in the instance and they all worked 👌

File: packages/grafana-ui/src/components/Link/Link.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { AnchorHTMLAttributes, forwardRef } from 'react';
-import { Link as RouterLink } from 'react-router-dom';
+import { Link as RouterLink } from 'react-router-dom-v5-compat';
```
Comment (jackw) [inline packages/grafana-ui/src/components/Link/Link.tsx:2]: Does grafana/ui have this package as a dependency?

**Comment by jackw** [comment]
It looks like `<TextLink>` is used by surrealdb datasource [here](https://github.com/grafana/surrealdb-datasource/blob/639752e5bcc4dcfb29f9e54915cd947e3eb1f3a4/src/components/ConfigEditor.tsx). I think it'd be worth double checking this doesn't cause any bother there before merging.

File: packages/grafana-ui/src/components/Link/Link.tsx (hunk 1 above)
Comment (Clarity-89) [reply packages/grafana-ui/src/components/Link/Link.tsx:2]: Good catch! Will replace `react-router-dom` with this package.

**Comment by Clarity-89** [comment]
> * https://ephemeral1511182192976clarity.grafana-dev.net

@jackw I've confirmed that that link component works correctly.

**Pull request merged**